jujugui/static/gui/src/app/components/deployment/summary/summary.js

```diff
@@ -43,6 +43,55 @@ YUI.add('deployment-summary', function() {
       validateForm: React.PropTypes.func.isRequired
     },
 
+    getInitialState: function() {
+      return {
+        regions: [],
+        activeRegion: this.props.deploymentStorage.region
+      };
+    },
+
+    componentWillMount: function() {
+      // If the model has already been deployed then we already
+      // have all the information we need.
+      var props = this.props;
+      if (props.modelCommitted) { return; }
+      // Because the user may have selected a template from the choose-cloud
+      // component we won't have any of the location data for the template
+      // to be able to properly select the correct region or cloud from the
+      // drop down list.
+      var deploymentStorage = props.deploymentStorage;
+      var templateName = deploymentStorage.templateName;
+      props.jem.listTemplates((error, credentials) => {
+        if (error) {
+          console.error('Unable to list templates', error);
+          return;
+        }
+        // The credentials are returned in an array so we need to loop through
+        // the list to find the one with the proper path.
+        credentials.forEach(credential => {
```

Comment:
You could use `credentials.some(...` here to break the loop when it finds the matching template name.